src/tensor.cpp

```diff
@@ -865,6 +876,1200 @@ void from_json(const nlohmann::json& j, TensorDescriptor& descriptor)
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[2]));
+            });
+
+            break;
+        }
+        case 4: {
+            visit_float(dataTypey, [&](auto as_float) {
+                kernel(x,
+                       *as_float(alpha),
+                       y,
+                       *as_float(beta),
+                       static_cast<unsigned>(Xoffset),
+                       static_cast<unsigned>(Yoffset),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[0]),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[1]),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[2]),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[3]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[0]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[1]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[2]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[3]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[0]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[1]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[2]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[3]));
+            });
+
+            break;
+        }
+        case 5: {
+            visit_float(dataTypey, [&](auto as_float) {
+                kernel(x,
+                       *as_float(alpha),
+                       y,
+                       *as_float(beta),
+                       static_cast<unsigned>(Xoffset),
+                       static_cast<unsigned>(Yoffset),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[0]),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[1]),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[2]),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[3]),
+                       static_cast<unsigned>(xDesc_flat.GetStrides()[4]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[0]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[1]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[2]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[3]),
+                       static_cast<unsigned>(yDesc_flat.GetStrides()[4]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[0]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[1]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[2]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[3]),
+                       static_cast<unsigned>(yDesc_flat.GetLengths()[4]));
+            });
+
+            break;
+        }
+        default: assert(false);
+        }
+    }
+}
+
+void OpTensor(Handle& handle,
```

Comment:
```suggestion
void OpTensor(const Handle& handle,
```

All other tensor functions works with const handle. Just like the old version of the OpTensor function.
The reason as I can see this function 

https://github.com/ROCm/MIOpen/blob/370e378a984cb9d6cd870d1b8fd7a346d4325a08/src/include/miopen/find_solution.hpp#L470-L477

Which mistakenly non-const , but that's not necessary.

https://github.com/ROCm/MIOpen/blob/370e378a984cb9d6cd870d1b8fd7a346d4325a08/src/include/miopen/find_solution.hpp#L441-L447

Reply:
And this should be resolved in https://github.com/ROCm/MIOpen/pull/3402.
which automatically resolves the merge conflict.